PWGHF/Core/HFMLResponse.h

```diff
@@ -0,0 +1,225 @@
+  TPCAndTOF = 0,
+  TPCOnly,
+  TOFOnly
+};
+
+template <typename T1>
+/// Tag track PID status (TPC and TOF, only TPC, only TOF)
+/// \param track is the track
+/// \return tag of PID status of the track
+uint8_t tagPID(const T1& track)
+{
+  uint8_t tag{0};
+  if (track.hasTPC() && track.hasTOF()) {
+    SETBIT(tag, PIDStatus::TPCAndTOF);
+  } else if (track.hasTPC() && !track.hasTOF()) {
+    SETBIT(tag, PIDStatus::TPCOnly);
+  } else {
+    SETBIT(tag, PIDStatus::TOFOnly);
+  }
+  return tag;
+}
+
+/// Combine TPC and TOF nSigma
+/// \param nSigTPC is the number of sigmas for TPC
+/// \param nSigTOF is the number of sigmas for TOF
+/// \param tagPID is the tag on PID status of the track
+/// \return combined nSigma value
+template <typename T1, typename T2>
+T1 getCombinedNSigma(const T1& nSigTPC, const T1& nSigTOF, const T2& tagPID)
+{
+  if (TESTBIT(tagPID, PIDStatus::TPCAndTOF)) {
+    return std::sqrt((nSigTPC * nSigTPC + nSigTOF * nSigTOF) / 2);
+  }
+  if (TESTBIT(tagPID, PIDStatus::TPCOnly)) {
+    return std::abs(nSigTPC);
+  }
+
+  return std::abs(nSigTOF); // if (TESTBIT(tagPID, PIDStatus::TOFOnly))
+}
+
+template <typename T = float>
+class HFMLResponse
+{
+ public:
+  /// Default constructor
+  HFMLResponse() = default;
+  /// Default destructor
+  virtual ~HFMLResponse() = default;
+
+  /// Configure class instance (import configurables)
+  /// \param binsLimits is a vector containing bins limits
+  /// \param cuts is a LabeledArray containing selections per bin
+  /// \param cutDir is a vector telling whether to reject score values greater or smaller than the threshold
+  /// \param nClasses is the number of classes for each model
+  /// \param paths is a vector of onnx model paths
+  void config(const std::vector<double>& binsLimits, const o2::framework::LabeledArray<double>& cuts, const std::vector<int>& cutDir,
+              const uint8_t& nClasses, const std::vector<std::string>& paths)
+  {
+    mBinsLimits = binsLimits;
+    mCuts = cuts;
+    mCutDir = cutDir;
+    mNClasses = nClasses;
+    mNModels = binsLimits.size() - 1;
+    mModels = std::vector<o2::ml::OnnxModel>(mNModels);
+    mPaths = paths;
+  }
+
+  /// Configure class instance (import configurables) when accessing model via CCDB
+  /// \param binsLimits is a vector containing bins limits
+  /// \param cuts is a LabeledArray containing selections per bin
+  /// \param cutDir is a vector telling whether to reject score values greater or smaller than the threshold
+  /// \param nClasses is the number of classes for each model
+  void config(const std::vector<double>& binsLimits, const o2::framework::LabeledArray<double>& cuts, const std::vector<int>& cutDir,
+              const uint8_t& nClasses)
+  {
+    mBinsLimits = binsLimits;
+    mCuts = cuts;
+    mCutDir = cutDir;
+    mNClasses = nClasses;
+    mNModels = binsLimits.size() - 1;
+    mModels = std::vector<o2::ml::OnnxModel>(mNModels);
+    mPaths = std::vector<std::string>(mNModels);
```

Comment:
This could be done also in the first version of `config` if `paths` is empty.

Reply:
Indeed, I only need one `configure` method then, thanks.